src/event_object_movement.c

```diff
@@ -2360,9 +2390,10 @@ bool32 CheckMsgCondition(const struct MsgCondition *cond, struct Pokemon *mon, u
         return (multi == cond->data.bytes[0] || multi == cond->data.bytes[1]);
     case MSG_COND_MUSIC:
         return (cond->data.raw == GetCurrentMapMusic());
-    // Added on `lighting` branch
-    // case MSG_COND_TIME_OF_DAY:
-    //     break;
+    case MSG_COND_TIME_OF_DAY:
+        // Must match time of day, have natural light on the map,
+        // and not have weather that obscures the sky
+        return (cond->data.raw == gTimeOfDay && MapHasNaturalLight(gMapHeader.mapType) && GetCurrentWeather() < WEATHER_RAIN);
```

Comment:
If this has vanilla precedent, are we sure rain prevents this from triggering and that eg. fog doesn't? If it is just rain, aren't we missing heavy rain? I assume the Kyogre/Groudon weathers would also qualify for the time of day being "unclear"?

Reply:
This check wasn't really clear on what it did. It assumes that all weather before rain are clear sky weathers, so I'm changing it to check specifically for `WEATHER_NONE/SUNNY_CLOUDS/SUNNY`.